src/integer/mat_z/trace.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains the implementation of the `trace` function.
+
+use flint_sys::fmpz_mat::fmpz_mat_trace;
+
+use super::MatZ;
+use crate::{
+    error::MathError,
+    integer::Z,
+    traits::{GetNumColumns, GetNumRows},
+};
+
+impl MatZ {
+    /// Returns the trace of a matrix and an error,
+    /// if the matrix is not square.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::MatZ;
+    /// use qfall_math::traits::*;
+    /// use std::str::FromStr;
```

Comment:
```suggestion
    /// use qfall_math::integer::MatZ;
    /// use std::str::FromStr;
```

this import is not used